navigation-material/src/main/java/com/google/accompanist/navigation/material/BottomSheetNavigator.kt

```diff
@@ -132,9 +141,9 @@ public fun rememberBottomSheetNavigator(
 @ExperimentalMaterialNavigationApi
 @OptIn(ExperimentalMaterialApi::class)
 @Navigator.Name("BottomSheetNavigator")
-public class BottomSheetNavigator(
+class BottomSheetNavigator internal constructor(
```

Comment:
Can you explain why this is in an `internal` constructor now? If this is intentional, make sure to add a comment stating that should only be constructed via `rememberBottomSheetNavigator` to the KDocs if that is indeed the only entry point supported.

Reply:
Good call, will point it out and softly deprecate it for now. Folks are constructing navigators with custom sheet states which is not something we want to support, and `skipHalfExpanded` support will likely require changing the constructor so I'd like to not have to maintain binary compatibility for this in the future.

Reply:
Decided not to deprecate it for the time being. I saw some folks pass a sheet state with `skipHalfExpanded = true` which is a valid use case imo